plugin/keyring_vault/vault_credentials.cc

```diff
@@ -0,0 +1,18 @@
+#include "vault_credentials.h"
+
+namespace keyring
+{
+
+const Secure_string& get_credential(const Vault_credentials &credentials, const Secure_string &key)
+{
+  static Secure_string empty_value;
```

Comment:
Local static object initialization is not thread safe in c++03. Magic statics appeared starting from c++11. So, move `empty_value` definition outside of the function body